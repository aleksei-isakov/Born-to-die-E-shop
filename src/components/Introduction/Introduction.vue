<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
    <p>
      For a guide and recipes on how to configure / customize this project,<br />
      check out the
      <a href="https://cli.vuejs.org" target="_blank" rel="noopener"
        >vue-cli documentation</a
      >.
    </p>
    <p>
      The project uses Material UI design. To start using its components <br />
      check out the official
      <a
        href="https://www.creative-tim.com/vuematerial"
        target="_blank"
        rel="noopener"
        >Vue Material documentation</a
      >.
    </p>
  </div>
</template>

<script>
export default {
  name: 'HelloWorld',
  props: {
    msg: String
  }
};
</script>

<style>
@import './Introduction.module.scss';
</style>
