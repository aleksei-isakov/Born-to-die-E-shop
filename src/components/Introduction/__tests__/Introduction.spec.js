import { shallowMount } from '@vue/test-utils';
import Introduction from '../Introduction.vue';

describe('Introduction.vue', () => {
  it('renders props.msg when passed', () => {
    const msg = 'new message';
    const wrapper = shallowMount(Introduction, {
      propsData: { msg }
    });

    expect(wrapper.text()).toMatch(msg);
  });

  it('renders a valid snapshot', () => {
    const wrapper = shallowMount(Introduction, {
      propsData: { msg: 'some message' }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
