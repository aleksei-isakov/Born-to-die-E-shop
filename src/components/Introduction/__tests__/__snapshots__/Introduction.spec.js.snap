// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction.vue renders a valid snapshot 1`] = `
<div class="hello">
  <h1>some message</h1>
  <p>
    For a guide and recipes on how to configure / customize this project,<br>
    check out the
    <a href="https://cli.vuejs.org" target="_blank" rel="noopener">vue-cli documentation</a>.
  </p>
  <p>
    The project uses Material UI design. To start using its components <br>
    check out the official
    <a href="https://www.creative-tim.com/vuematerial" target="_blank" rel="noopener">Vue Material documentation</a>.
  </p>
</div>
`;
